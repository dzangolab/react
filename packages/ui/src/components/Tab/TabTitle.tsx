import React from "react";

type Properties = {
  title: string;
  eventKey: number;
  selected: number;
  setSelectedTab: (index: number) => void;
};

const TabTitle: React.FC<Properties> = ({
  title,
  selected,
  setSelectedTab,
  eventKey,
}) => {
  return (
    <li>
      <button
        className={`tab-title ${selected === eventKey ? "active" : ""}`}
        onClick={() => setSelectedTab(eventKey)}
      >
        {title}
      </button>
    </li>
  );
};

export default TabTitle;
