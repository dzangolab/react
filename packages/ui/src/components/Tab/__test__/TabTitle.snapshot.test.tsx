import React from "react";
import renderer, { create } from "react-test-renderer";
import { expect, test } from "vitest";

import TabTitle from "../TabTitle";

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON();

  expect(result).toBeDefined();
  expect(result).not.toBeInstanceOf(Array);

  return result as renderer.ReactTestRendererJSON;
}

test("Component matches snapshot", () => {
  const component = create(
    <TabTitle
      title="tab title"
      eventKey={1}
      selected={1}
      // eslint-disable-next-line @typescript-eslint/no-empty-function
      setSelectedTab={() => {}}
    />
  );
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
