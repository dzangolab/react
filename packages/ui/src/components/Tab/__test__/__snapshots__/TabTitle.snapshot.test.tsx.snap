// Vitest Snapshot v1

exports[`Component matches snapshot 1`] = `
<li>
  <button
    className="tab-title active"
    onClick={[Function]}
  >
    tab title
  </button>
</li>
`;
